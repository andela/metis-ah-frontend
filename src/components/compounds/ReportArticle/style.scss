* {
  outline: none;
}

.reportArticle {
  position: fixed;
  width: 100%;
  height: 100vh;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  display: grid;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.7);
  z-index: 900;

  .reportArticleModal {
    display: grid;
    justify-items: center;
    position: absolute;
    top: 50%;
    left: 50%;
    max-height: 90%;
    transform: translate(-50%, -50%);
    background: white;
    padding-bottom: 2em;

    .report_header {
      padding: 10px 15px;
      background: #9ddec7;
      width: 100%;
      text-align: center;
      margin-bottom: 10px;
      font-size: 1.2em;
      font-weight: bold;
    }

    .reportReasons {
      margin-bottom: 15px;
    }

    li {
      list-style: none;
      color: rgba(0, 0, 0, 0.54);
      margin: 5px;
      border-radius: 4px;
      cursor: pointer;

      input[type='radio'] {
        margin-right: 10px;
      }
    }

    .buttonGroup {
      display: grid;
      grid-template-columns: 1fr 1fr;
      justify-items: center;
      grid-gap: 8px;

      .submitButton {
        justify-self: start;
        border: 1px solid #4dd6a5;
        color: #4dd6a5;

        &:hover,
        &:active {
          color: white;
          background: #4dd6a5;
          border: 1px solid #4dd6a5;
        }
      }

      .cancelButton {
        justify-self: end;
        border: 1px solid #c9c9c9;
        color: #888;

        &:hover,
        &:active {
          color: white;
          background: #888;
          border: 1px solid #888;
        }
      }
    }

    .reportForm {
      width: 100%;
      padding: 0 2em;
    }

    .reportBody {
      resize: none;
      min-height: 150px;
      border: 1px solid #e0e0e0;
      border-radius: 4px;
      padding: 10px;
      font-size: 0.9em;
      justify-self: stretch;
      outline: none;
      width: 100%;

      &::placeholder {
        color: #b6b6b6;
      }
    }
  }
}

.closeButton {
  width: 20px;
  height: auto;
  justify-self: end;
  align-self: start;
  margin-right: 20px;
  cursor: pointer;
  margin-top: 20px;

  svg {
    fill: rgba(230, 229, 229, 0.54);
  }
}

button {
  background: none;
  border-radius: 15px;
  padding: 5px 20px;
  font-size: 14px;
  cursor: pointer;
  transition: 0.2s background ease-in;
}

@media screen and (max-width: 500px) {
  .reportArticle {
    left: 0;

    .reportArticle {
      width: 100%;

      .reportArticleModal {
        top: 0;
        left: 0;

        .reportForm {
          width: 100%;
          padding-left: 0.5em;
          padding-right: 0.5em;
        }
      }

    }
  }
}

.reportReasons {
  width: 100%;
  padding: 0 2em;
  font-size: 0.9em;
}

label {
  font-size: 14px;
  color: #555;
}