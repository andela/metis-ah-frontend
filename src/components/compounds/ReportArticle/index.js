import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import ReactDOM from "react-dom";
import Alert from "Utils/helpers/alert";

import {
  reportArticle,
  clearMessage
} from "../../../store/actions/reportArticle";
import "./style.scss";

const modalRoot = document.getElementById("modal");

class ReportArticle extends Component {
  el = document.createElement("div");

  state = {
    reasonForReport: "",
    description: ""
  };

  componentDidMount() {
    modalRoot.appendChild(this.el);
  }

  componentWillUnmount() {
    const { clearResponseMessage } = this.props;
    clearResponseMessage();
    modalRoot.removeChild(this.el);
  }

  handleChange = event => {
    this.setState({
      reasonForReport: event.target.value
    });
  };

  updateReportBody = event => {
    this.setState({
      description: event.target.value
    });
  };

  handleSubmitReport = event => {
    event.preventDefault();
    const { makeReport, articleId, closeModal } = this.props;
    const { reasonForReport, description } = this.state;

    if (reasonForReport === "" || reasonForReport === undefined) {
      return Alert.error("Please select reason for the report");
    }
    if (description === "" || description === undefined) {
      return Alert.error("Please add a short description of the report");
    }

    makeReport(articleId, reasonForReport, description).then(response => {
      if (response) {
        closeModal();
      }
    });
  };

  render() {
    const { closeModal, loading, message, error } = this.props;
    const { reasonForReport } = this.state;

    return ReactDOM.createPortal(
      <Fragment>
        <div className="reportArticle">
          {error ? Alert.error(error) : null}
          {message ? Alert.success(message) : null}
          <span className="closeButton" onClick={closeModal}>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20">
              <path d="M10 8.586L2.929 1.515 1.515 2.929 8.586 10l-7.071 7.071 1.414 1.414L10 11.414l7.071 7.071 1.414-1.414L11.414 10l7.071-7.071-1.414-1.414L10 8.586z" />
            </svg>
          </span>
          <div className="reportArticleModal">
            <div className="report_header">
              <h2>Report Article</h2>
            </div>
            <div className="reportReasons">
              <li>
                <label>
                  <input
                    type="radio"
                    checked={reasonForReport === "Discrimination"}
                    onChange={this.handleChange}
                    value="Discrimination"
                  />
                  Discrimination
                </label>
              </li>
              <li>
                <label>
                  <input
                    type="radio"
                    checked={reasonForReport === "Plagiarism"}
                    onChange={this.handleChange}
                    value="Plagiarism"
                  />
                  Plagiarism
                </label>
              </li>
              <li>
                <label>
                  <input
                    type="radio"
                    checked={reasonForReport === "Sexual Content"}
                    onChange={this.handleChange}
                    value="Sexual Content"
                  />
                  Sexual Content
                </label>
              </li>
              <li>
                <label>
                  <input
                    type="radio"
                    checked={reasonForReport === "Offensive Language"}
                    onChange={this.handleChange}
                    value="Offensive Language"
                  />
                  Offensive Language
                </label>
              </li>
            </div>
            <div className="reportForm">
              <form onSubmit={this.handleSubmitReport}>
                <textarea
                  onChange={this.updateReportBody}
                  className="reportBody"
                  placeholder="Add a short description of the report"
                />
                <div className="buttonGroup">
                  <button
                    type="button"
                    className="cancelButton"
                    onClick={closeModal}
                  >
                    Cancel
                  </button>
                  <button
                    type="submit"
                    className="submitButton"
                    disabled={loading}
                  >
                    Submit
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </Fragment>,
      modalRoot
    );
  }
}

const mapStateToProps = state => {
  const { reportArticle } = state;
  const { message, error, loading } = reportArticle;
  return {
    message,
    error,
    loading
  };
};

const mapDispatchToProps = dispatch => ({
  makeReport(articleId, violation, description) {
    return dispatch(reportArticle(articleId, violation, description));
  },
  clearResponseMessage() {
    return dispatch(clearMessage());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ReportArticle);
