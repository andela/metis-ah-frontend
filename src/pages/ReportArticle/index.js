import React, { Component, Fragment } from "react";
import { connect } from "react-redux";

import reportArticle from "../../store/actions/reportArticle";
import './style.scss;'

const modalRoot = document.getElementById("modal");

class ReportArticle extends Component {
  el = document.createElement("div");
  
  componentDidMount() {
    modalRoot.appendChild(this.el);
  }

  componentWillUnmount() {
    modalRoot.removeChild(this.el);
  }

  reportArticle = (violation, description) => {
    const { makeReport } = this.reportArticle;
  }

  render() {
    return (
      <Fragment>
        <div className='report_article'>
          
        </div>
      </Fragment>
    )
  }
}

const mapStateToProps = state => {
  const { reportArticle } = state;
  const { message, error, loading } = reportArticle;
  return {
    message,
    error,
    loading
  };
};

const mapDispatchToProps = dispatch => ({
  makeReport(violation, description) {
    return dispatch(reportArticle(violation, description));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ReportArticle);
