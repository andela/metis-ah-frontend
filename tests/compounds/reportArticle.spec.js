import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import ReportArticle from "../../src/components/compounds/ReportArticle";

const mockStore = configureMockStore();
const store = mockStore();

const mockFunction = param => jest.fn;

const props = {
  makeReport: mockFunction,
  clearResponseMessage: mockFunction
};

describe("Report Article Component", () => {
  const wrapper = shallow(
    <Provider store={store}>
      <ReportArticle />
    </Provider>
  );

  it("should render the report article component", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
