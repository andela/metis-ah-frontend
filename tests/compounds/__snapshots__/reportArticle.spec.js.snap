// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Article Component should match snapshot 1`] = `<Connect(ReportArticle) />`;
